### ISSUE TITLE
Recover support to write properties for non stdClass objects 

### ISSUE THREAD

**Issue by ScreamingDev** [symfony/symfony#37026]
**Symfony version(s) affected**: 5.1 and lower

**Description**  
When generating entities (e.g. via nelmio/alice) it was long possible to write values into existing properties.
But now it seems like the property accessor does only allows this for instances of stdClass .

Possibly since this change: https://github.com/symfony/property-access/commit/0583535d2cfa597ccd6a8e75f6f3bf5a138a5541

So when a class like

```
class Foo {
}
```

has no property but an object gets a property later on like

```
$o = new Foo();
$o->bla = null;
```

then the PropertyAcessor is no longer able to write into this existing property
and instead throws the NoSuchPropertyException exception with a "Could not determine access type" message.

**Possible Solution**  

Drop the explicit check for stdClass 

```
} elseif (property_exists($object, $property)) {
            $object->$property = $value;
        }
```

see https://github.com/symfony/property-access/blob/0583535d2cfa597ccd6a8e75f6f3bf5a138a5541/PropertyAccessor.php#L509

**Comment by onehorsetown** [comment]
This exact same issue is present trying to read a dynamically added property.   It worked fine in 5.0, but broke in 5.1

https://github.com/symfony/symfony/blob/367aa1df6caa2c739a58165ead81e7fe50e6599e/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L438

So something as simple as this no longer works:

```
class Person  {
    public function __construct(string $name) {
        $this->name = $name;
    }
}

$accessor = PropertyAccess::createPropertyAccessor();

$person = new Person("Joe Consumer");
$name = $accessor->getValue($person, 'name');   // KABOOM - Can't get a way to read the property "name" in class...
```

One solution is to declare the property in the class definition, but this does not solve the issue for dynamically added properties.

**Comment by andreyserdjuk** [comment]
Sure property_exists() can show that property exists but it can be inaccessible, however, it was described in earlier PropertyAccessor versions comments. My try to fix this #37622

**Comment by garak** [comment]
There's still a check for stdClass in https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L526 so I think this issue should be re-opened.

I'm facing the same problem, by trying to write a dynamic property I get `Could not determine access type for property "myDynamicProperty" in class MyClass`
